packages/bot-arbitrage/src/arbitarger.ts

```diff
@@ -0,0 +1,148 @@
+import { ArbParams, Market, MarketWithToken, Method } from "./types";
+import { ArbParamsStruct, MgvArbitrage } from "./types/typechain/MgvArbitrage";
+import { Mangrove, typechain, MgvToken } from "@mangrovedao/mangrove.js";
+
+export const activateTokens = async (
+  mgv: Mangrove,
+  tokens: MgvToken[],
+  arbitragerContract: MgvArbitrage
+) => {
+  const values = await Promise.all(
+    tokens.map((token) => {
+      return token.allowance({
+        owner: arbitragerContract.address,
+        spender: mgv.address,
+      });
+    })
+  );
+
+  const tokenToActivate = values
+    .map((value, index) => ({ value, index }))
+    .filter((value) => value.value.eq(0))
```

Comment:
We assume that not 0 is infinite approval ?

Reply:
Yes